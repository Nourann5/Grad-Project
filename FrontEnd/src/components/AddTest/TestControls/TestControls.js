import React, { useEffect } from 'react'
import styles from './TestControls.module.css'
import {ReactComponent as InvoLogo} from 'assets/icons/InvoLogo.svg'

import {ReactComponent as QuestionIcon} from 'assets/icons/question.svg'
import {ReactComponent as PlusIcon} from 'assets/icons/plus.svg'
import {ReactComponent as PlayTest} from 'assets/icons/playTest.svg'
import {ReactComponent as DeleteXIcon} from 'assets/icons/deleteX.svg'
import {ReactComponent as MenuIcon} from 'assets/icons/menu.svg'
import {ReactComponent as AminusIcon} from 'assets/icons/aMinus.svg'
import {ReactComponent as TimerIcon} from 'assets/icons/timer.svg'
import TestSubmitModal from '../TestSubmitModal/TestSubmitModal'
import { Link } from 'react-router-dom'
    // console.log('QuestionConfiguration')

function TestControls({addQuestion,deleteQuestion ,questionslCount ,endExam,toggleSideNavBar}) {
    const EndExamButtonComp= ({openSubmitModal})=>{
        return(
            <button type='button' className={`${styles["test__control-end-task"]} btn mx-auto`} onClick={openSubmitModal}>
                Add Questions
            </button>
        )
    }

  return (
        <div className={styles["test__controls"]}>
            <div className={styles["test__controls__cont"]}>

                <button 
                    type='button' 
                    className={`${styles["test__control-button"]} ${styles["test__control-button--rotate"]} ${styles["test__control-button--hidden"]} btn`}
                    onClick={toggleSideNavBar}
                >
                    <MenuIcon className={styles["test__control-button-icon"]}/>
                </button>
                <Link to='/portal/preview-add-test' target='_blank'><PlayTest className={styles["test__control-button-icon"]}/></Link>
                
                {/* <button type='button' className={`${styles["test__control-button"]} btn`}>
                    <APlusIcon className={styles["test__control-button-icon"]}/>
                </button>
                <button type='button' className={`${styles["test__control-button"]} btn`}>
                    <AminusIcon className={styles["test__control-button-icon"]}/>
                </button> */}
            </div>

            <TestSubmitModal EndExamButtonComp={<EndExamButtonComp/>} endExam={endExam}/>
            <div className={styles['add-test__control-wrapper']}>
                {
                    questionslCount>1&&
                        <button type="button" className={styles['add-test__delete-button']} onClick={deleteQuestion}>
                            <DeleteXIcon className={styles['add-test__delete-button-icon']}/>
                        </button>
                }
                <button type="button" className={styles['add-test__add-button']} onClick={addQuestion}>
                    <PlusIcon className={styles['add-test__add-button-icon']}/>
                </button>
            </div>

            {/* <div className={styles["test__timer-cont"]}>
                <TimerIcon className={styles["test__control-button-icon"]}/>
                <span className={styles["test__show-time js-test__show-time-min"]}>
                20
                </span>:
                <span className={styles["test__show-time js-test__show-time-sec"]}>
                85
                </span>
            </div> */}
        </div>
  )
}

export default TestControls