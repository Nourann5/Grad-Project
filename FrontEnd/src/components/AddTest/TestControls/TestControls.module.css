.test-controls__logo{
    width: 150px;
    margin-right: 20px;
}
.test__controls{
    background-color:white;
    border-bottom: 1px solid #00000012;
    /* box-shadow: 0 2px 6px #00000012; */
    display: flex;
    align-items: center;
    height: 60px;
    padding: 0 20px;
    /* margin-top: 20px; */
    margin-bottom: 30px;
}
.test__controls__cont{
    position: absolute;
}
.test__control-button{
    margin-right: 15px;
}
.test__control-button-icon{
    width:25px;
}
.test__control-button-icon path{
    fill:var(--main-color)
}
.test__control-button--rotate{
    transform:rotate(180deg);
}
.test__control-end-task{
    background-color: var(--main-color);
    color: white;
    font-size: 13px;
    width: 160px;
    height: 40px;
    transition:var(--transition);
    font-family: 'PoppinsSemiBold';
}
.test__control-end-task:hover ,.test__control-end-task:focus{
    background-color: var(--main-color);
    color:white;
    opacity: .8;
}
.test__timer-cont{
    display: flex;
    align-items: center;
    position: absolute;
    right:90px
}
.test__timer-cont svg{
    margin-right: 10px;
}
.test__timer-cont span{
    font-family: 'PoppinsBold';
    color: var(--main-color);
    width: 20px;
}
.add-test__add-button{
    background-color: transparent;
    border: 3px solid var(--main-color)!important;
    border-radius: 50%;
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
}
.add-test__delete-button{
    background-color: transparent;
    border: 3px solid red!important;
    border-radius: 50%;
    width: 40px;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: center;
    margin-right: 10px;
}
.add-test__add-button-icon{
    width: 20px;
    height: 20px;
    color: var(--main-color);
}
.add-test__delete-button-icon{
    width: 20px;
    height: 20px;
    color: red;
}
.add-test__control-wrapper{
    position: absolute;
    right:50px;
    display: flex;
    align-items: center;
    justify-content: center;
}

@media (max-width:992px){
    .test__controls{
        padding:5px
    }
    .add-test__control-wrapper{
        right:10px;
    }
    .test__control-button{
        margin:0
    }
    .test__control-button--hidden{
        display:none
    }
    /* .test__control-end-task{
        order: 1;
        margin-bottom: 10px
    }
    .test__controls__cont{
        position: initial;
        order: 3;
    }
    .test__timer-cont{
        position: initial;
        order: 2;
    } */
    
}
@media (max-width:769px){
    .add-test__delete-button{
        margin-right: 5px;
    }
   
}