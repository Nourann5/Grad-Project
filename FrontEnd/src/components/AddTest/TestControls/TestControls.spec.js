import {render , renderHook, screen} from '@testing-library/react'
import { BrowserRouter, Route } from 'react-router-dom'
import { FormProvider, useForm } from 'react-hook-form'
import TestControls from './TestControls'

const methods = renderHook(()=> useForm())

describe('Test the Add Test ', () => { 
    it('TestControls',()=>{
        const TestInput = () => {
            const methods = useForm();
            return (
                <BrowserRouter>
                    <FormProvider {...methods}>
                        <TestControls />
                    </FormProvider>
                </BrowserRouter>
            );
        };
        const tree =  render(<TestInput/>)
        expect(tree).toMatchSnapshot()
    })
})