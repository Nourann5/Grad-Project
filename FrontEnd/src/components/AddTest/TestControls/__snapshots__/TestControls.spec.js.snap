// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Add Test  TestControls 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
    <div>
      <div
        class="test__controls"
      >
        <div
          class="test__controls__cont"
        >
          <button
            class="test__control-button test__control-button--rotate test__control-button--hidden btn"
            type="button"
          >
            <svg
              class="test__control-button-icon"
            >
              menu.svg
            </svg>
          </button>
          <a
            href="/portal/preview-add-test"
            target="_blank"
          >
            <svg
              class="test__control-button-icon"
            >
              playTest.svg
            </svg>
          </a>
        </div>
        <button
          class="test__control-end-task btn mx-auto"
          type="button"
        >
          Add Questions
        </button>
        <div
          class="add-test__control-wrapper"
        >
          <button
            class="add-test__add-button"
          >
            <svg
              class="add-test__add-button-icon"
            >
              plus.svg
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="test__controls"
    >
      <div
        class="test__controls__cont"
      >
        <button
          class="test__control-button test__control-button--rotate test__control-button--hidden btn"
          type="button"
        >
          <svg
            class="test__control-button-icon"
          >
            menu.svg
          </svg>
        </button>
        <a
          href="/portal/preview-add-test"
          target="_blank"
        >
          <svg
            class="test__control-button-icon"
          >
            playTest.svg
          </svg>
        </a>
      </div>
      <button
        class="test__control-end-task btn mx-auto"
        type="button"
      >
        Add Questions
      </button>
      <div
        class="add-test__control-wrapper"
      >
        <button
          class="add-test__add-button"
        >
          <svg
            class="add-test__add-button-icon"
          >
            plus.svg
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
